evennia/contrib/slots_handler/slots.py

```diff
@@ -0,0 +1,385 @@
+                return new
+            else:
+                array.update(to_add)
+                return array
+
+    def delete(self, slots):
+        """
+        This will delete slots from an existing array.
+        WARNING: If you have anything attached in slots when they are removed,
+        the slots' contents will also be removed. This function will return a
+        dict of any removed slots and their contents, so it can act as a pop(),
+        but if you don't catch that data, it WILL be lost.
+
+        Args:
+            slots (list or dict): Slot categories or individual slots to
+                delete.
+        """
+
+        if not isinstance(slots, (dict, _SaverDict, list, _SaverList)):
+            return StatMsg(False, "You have to declare slots in the form "
+                           "`{key: [values]}`, or categories in the form "
+                           "`[values]`.")
+
+        arrays = {a.key: a for a in self.obj.slots.all(obj=True)}
+        deleted = {}
+        for name in slots:
+            self.is_name_valid(name)
+            existing = arrays.get(name, False)
+            del_temp = {}
+            if not existing:
+                # If the named array isn't there, skip to the next one.
+                break
+            array = existing.value
+
+            if isinstance(slots, (list, _SaverList)):
+                # If the input is a list, it is interpreted as a list of
+                # category names and all slots are deleted.
+                deleted.update({name: array})
+                self.obj.attributes.delete(name, category="slots")
+            elif isinstance(slots, (dict, _SaverDict)):
+                # If the input is a dict, only the specific slots indicated
+                # will be deleted.
+                self.defrag_nums(existing.key)  # Just in case.
+                named = {k: v for k, v in array.items()
+                         if isinstance(k, str)}
+                numbered = {k: v for k, v in array.items()
+                            if isinstance(k, int)}
+                to_del = [s for s in slots[name] if isinstance(s, str)]
+                highest = sorted(numbered.keys() + [0])[-1]
+                del_num = sum([n for n in slots[name] if isinstance(n, int)])
+                to_del = to_del + [i for i
+                                   in range(highest, highest - del_num, -1)]
+
+                del_temp = {d: array.pop(d) for d in to_del}
+                deleted.update({name: del_temp})
+
+        return deleted
+
+    def attach(self, target, slots=None):
+        """
+        Attempt to attach the target in all slots it consumes. Optionally, the target's slots may be overridden.
+
+        Args:
+            target (object): The object to be attached.
+            slots (list or dict, optional): If slot instructions are given,
+                this will completely override any slots on the object.
```

Comment:
A `Returns:` section is missing (it's missing all over, so won't point those out henceforth). An `Examples:` section is optional but may be useful too here.

Reply:
I'm not going to include `Examples:` in the docstrings because I feel like they would take up too much space and I can include a more thorough tutorial in the `README`.